### PR TITLE
[phi] Migrate update_loss_scaling XPU kernel to PHI.

### PR DESCRIPTION
**Pull request by jiahy0825**
### PR types
Others

### PR changes
Others

### Describe
Migrate update_loss_scaling XPU kernel to PHI.

### CODE REVIEW

File: paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    T* out_data = dev_ctx.template Alloc<T>(out);
+    int num = out->numel();
+    if (cpu_found_inf_data) {
+      VLOG(1) << "-- UpdateLossScaling: Find infinite grads. --";
+      int r = 0;
+      r = xpu::constant(dev_ctx.x_context(),
+                        reinterpret_cast<XPUTyp*>(out_data),
+                        num,
+                        XPUTyp(0.0));
+      PADDLE_ENFORCE_EQ(r,
```
Comment (helen88) [inline paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc:76]: PADDLE_ENFORCE_XDNN_SUCCESS 请用这个宏

File: paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <cstring>
+#include <string>
+#include <vector>
+
+#include "paddle/phi/kernels/amp_kernel.h"
```
Comment (chenwhql) [inline paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc:19]: cc中这个头文件在最前面

Comment (chenwhql) [reply paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc:19]: 头文件和cc文件命名保持一致

File: paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+#include "paddle/fluid/memory/memcpy.h"
+#include "paddle/phi/backends/xpu/xpu_context.h"
+#include "paddle/phi/common/amp_type_traits.h"
+#include "paddle/phi/common/float16.h"
+#include "paddle/phi/common/scalar.h"
```
Comment (chenwhql) [inline paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc:25]: scalar.h头文件中有了，是不是不需要include了

File: paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc (hunk 1 above)
Comment (jiahy0825) [reply paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc:76]: Done

File: paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc (hunk 3 above)
Comment (jiahy0825) [reply paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc:25]: Done

File: paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc (hunk 2 above)
Comment (jiahy0825) [reply paddle/phi/kernels/xpu/update_loss_scaling_kernel.cc:19]: Done

**Comment by jiahy0825** [comment]
`update_loss_scaling` and `check_finite_and_unscale` are both amp ops.
Migrate them through one PR: [[phi] Migrate amp XPU kernels to PHI](https://github.com/PaddlePaddle/Paddle/pull/45569)

**Pull request closed without merging**